docs/DeveloperGuide.md

```diff
@@ -1,34 +1,522 @@
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/Architecture.png" alt="Architecture diagram of PlanNUS"/>
+</div>
+
+The ***Architecture Diagram*** given above explains the high-level design of PlanNUS. Below is a quick overview of each component.
+
+
+
+### Overview
+
+#### PlanNus
+
+`PlanNus` class contains the `main` and `run` method, which is responsible for
+
+  * At launch
+
+       * Loading all modules for AY2020/21 into PlanNUS
+
+       * Loading previous save file into PlanNUS if available
+
+       * Creation of entry point to available apps in PlanNUS
+
+  * While running
+
+       * Continuously prompt user for app selection
+
+  * At shut down
+
+       * Saving of user data into save file
+       * Clean up methods where necessary
+
+
+
+#### Global, Ui, Parser, Storage, Apps
+
+* The `global` package contains classes, exceptions and objects that are required across the whole app. 
+* The `ui` package contains the class that is responsible for sharing one `scanner` class across the whole app to prevent multiple IO streams
+* The `parser` package contains the class that handles user's app selection
+* The `storage` package handles loading and saving of user's data to a save file.
+* Packages for Available apps such as Academic Planner and CAP Calculator are stored within `apps` package
+
+
+
+### Project Structure
+
+Each package in the PlanNUS as given above follows the following file structure where applicable:
+
+* A functional class that acts as the entry point to that module
+* A parser class that parses user input into executable commands by PlanNUS
+
+* `commands`: A package that handles all executable commands given by parser
+* `commons`: A package with the utilities and shared classes across the parent package
+* `exceptions`: A package to handle all exceptions thrown across the parent package
+
+The interaction within each package should ideally be as shown below.
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/Project_structure.png" alt="Architecture diagram for ideal project structure in PlanNUS"/>
+</div>
+*Note that while this is the ideal case, packages such as* `global`, `parser` *and* `ui` *might not strictly follow this structure due to these package serving a different function altogether (Refer to the sections below for more details.)*
+
+### Life cycle of PlanNUS
+
+The [*sequence diagram*](#sequence-diagram) below shows how different packages and classes interact with each other throughout the whole [life cycle](#life-cycle) of PlanNUS.
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/Packages_Interaction.png" alt="Sequence diagram for lifecycle of PlanNUS"/>
+</div>
+### Details
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/Details_architecture.png" alt="Details architecture diagram of PlanNUS"/>
+</div>
```

Comment:
I'm a little bit confused about the relation between Apps and PlanNUS because in the previous diagram "Architecture" there is no connection between Apps and other components and in this "Details" diagram there is no connection neither, maybe you can explain more about this?
<img width="431" alt="dg1" src="https://user-images.githubusercontent.com/60384579/97657283-741dc980-1aa4-11eb-8e72-60f32d13078a.png">